lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -765,7 +832,17 @@ namespace Js
         CleanupBeforeExecution();
 
         Arguments outArgs(CallInfo(CallFlags_Value, 0), nullptr);
-        return rootFunction->CallRootFunction(outArgs, scriptContext, true);
+        
+        Var ret = nullptr;
+        AUTO_NESTED_HANDLED_EXCEPTION_TYPE((ExceptionType)(ExceptionType_OutOfMemory | ExceptionType_JavascriptException));
+        ENTER_SCRIPT_IF(scriptContext, true, false, false, !scriptContext->GetThreadContext()->IsScriptActive(),
+        {
+            ret = rootFunction->CallRootFunction(outArgs, scriptContext, true);
```

Comment:
Added promise reject in latest iteration.

---
In reply to: [115362554](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115362554) [](ancestors = 115362554)